test/redis/rack/connection_test.rb

```diff
@@ -62,6 +62,11 @@ def setup
         conn.pooled?.must_equal false
         conn.store.to_s.must_match(/127\.0\.0\.1:6380 against DB 1$/)
       end
+
+      it "does not include nil options for the connection pool" do
+        conn = Connection.new
+        conn.pool_options.must_equal({})
```

Comment:
IMO this doesn't test much because `{}` is already `#compact`.

try actually setting the pool options explicitly to `nil`, like

```ruby
conn = Connection.new(pool_size: nil)

conn.pool_options.must_be_empty
```

Reply:
> this doesn't test much because `{}` is already `#compact`

It tests the exact case that fails for us. The test fails before the change and passes after.

Reply:
Updated

Reply:
ahh I see what you mean now, well thanks very much! looks good to me.